engine.go

```diff
@@ -1,36 +1,121 @@
 package main
 
 import (
+	"flag"
+	"fmt"
+	"io/ioutil"
 	"net/http"
+	"os/user"
+	"path"
 	"strings"
 
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/cloud"
+	"google.golang.org/cloud/datastore"
+
+	"github.com/coduno/piper/models"
 	"github.com/coduno/piper/runner"
 )
 
-func setupRunHandler(rh runner.RunHandler) http.HandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request) {
-		cors(w, r)
-		config := rh.Handle(w, r)
-
-		// TODO(flowlo): Find out whether Handle completed successfully
-		// to check.
+var ctx context.Context
 
-		res, err := config.Run()
+func init() {
+	user, err := user.Current()
+	if err != nil {
+		panic(err)
+	}
 
-		if err != nil {
-			http.Error(w, "docker: "+err.Error(), http.StatusInternalServerError)
-		}
+	fileName := path.Join(user.HomeDir, "config", "secret.json")
+	secret, err := ioutil.ReadFile(fileName)
+	if err != nil {
+		panic(err)
+	}
 
-		rh.Respond(w, r, res)
+	config, err := google.JWTConfigFromJSON(secret, datastore.ScopeDatastore, datastore.ScopeUserEmail)
+	if err != nil {
+		panic(err)
 	}
+
+	ctx = cloud.NewContext("coduno", config.Client(oauth2.NoContext))
```

Comment:
It is a bad idea to make `ctx` global. We should rather make the `http.Client` returned by `config.Client(...)` a global and [reuse it, as this is explicitly stated in the docs](http://golang.org/src/net/http/client.go?s=618:735#L30):

```go
var client http.Client

func init() {
	...
	client = config.Client(oauth2.NoContext)
}

func NewContext() (context.Context) {
	return cloud.NewContext("coduno", client)
}
```